### PR TITLE
feat: allow sentry to receive extra params

### PR DESCRIPTION
**Pull request by Ian2012**
## Description

This PR fixes the issue #247 

## Testing instructions

Set the variable `EOX_CORE_SENTRY_EXTRA_OPTIONS` to any custom param that sentry accepts

### CODE REVIEW

**Comment by mariajgrimaldi** [comment]
@Ian2012: can you give us an example of what to add to `EOX_CORE_SENTRY_EXTRA_OPTIONS`

File: eox_core/settings/production.py [hunk 1]
```diff
@@ -141,7 +145,7 @@ def plugin_settings(settings):  # pylint: disable=function-redefined
             integrations=[
                 DjangoIntegration(),
             ],
-
+            **sentry_extra_options,
```
Comment (mariajgrimaldi) [inline eox_core/settings/production.py:148]: can we move this to line 151?

Comment (mariajgrimaldi) [reply eox_core/settings/production.py:148]: what if `sentry_extra_options` is invalid?

**Comment by mariajgrimaldi** [comment]
Could you document these changes [here](https://github.com/eduNEXT/eox-core/#integrations-with-third-party-services)? Please

File: eox_core/settings/production.py (hunk 1 above)
Comment (Ian2012) [reply eox_core/settings/production.py:148]: What would be an invalid `sentry_extra_option`? What cases should I cover?

**Comment by Ian2012** [comment]
> @DonatoBD @DeimerM @Ian2012: this is a minor change but I'd like to test it remotely to check its behavior. Although, I don't know if creating a sentry project just for a test would be worth it. Have you tested this approach before?

@mariajgrimaldi We have not tested it, but we can do it in our stage environment

**Comment by mariajgrimaldi** [comment]
@Ian2012: please, do. Let us know when it's deployed so we can test it too. Also, it would be awesome if you could attach the testing instructions using sentry. Thanks!

File: eox_core/settings/production.py (hunk 1 above)
Comment (mariajgrimaldi) [reply eox_core/settings/production.py]: An option that's not handled by sentry_sdk.init, or does the method just ignores the options that don't recognize?

**Comment by Ian2012** [comment]
@mariajgrimaldi We performed a test using other variables and it works as expected.
```python
debug=True
```
I get a different error when trying to configure profiling:

EOX_CORE_SENTRY_EXTRA_OPTIONS = {
    "_experiments": { "profiles_sample_rate": 1.0, },
    "traces_sample_rate": 1.0
}
```
[sentry] DEBUG: [Profiling] Discarding profile because transaction is discarded.
--- Logging error ---
Traceback (most recent call last):
  File "/opt/pyenv/versions/3.8.12/lib/python3.8/logging/__init__.py", line 440, in format
    return self._format(record)
  File "/opt/pyenv/versions/3.8.12/lib/python3.8/logging/__init__.py", line 436, in _format
    return self._fmt % record.__dict__
KeyError: 'userid'
```

It seems it's a problem with the default logging format of edx-platform.

BTW, this functionality works and the above issue shouldn't be a blocker

File: eox_core/settings/production.py [hunk 2]
```diff
@@ -141,8 +145,8 @@ def plugin_settings(settings):  # pylint: disable=function-redefined
             integrations=[
                 DjangoIntegration(),
             ],
-
             # If you wish to associate users to errors (assuming you are using
             # django.contrib.auth) you may enable sending PII data.
-            send_default_pii=True
+            send_default_pii=True,
+            **sentry_extra_options
```
Comment (mariajgrimaldi) [inline eox_core/settings/production.py:151]: If I add a duplicated option (I did it by accident), instances won't be able to initialize. So can you please add a try-except for this?
image

Comment (mariajgrimaldi) [reply eox_core/settings/production.py:151]: Instances won't initialize either (at least workers) if the option doesn't exist:
image

**Review by mariajgrimaldi — CHANGES_REQUESTED** [review]

File: eox_core/settings/production.py (hunk 2 above)
Comment (Ian2012) [reply eox_core/settings/production.py]: I've added a try-except block that will ignore the `sentry_extra_options` param if sentry fails to initialize

Comment (Ian2012) [reply eox_core/settings/production.py]: What do you think? @mariajgrimaldi

Comment (Ian2012) [reply eox_core/settings/production.py]: Just a a reminder @mariajgrimaldi

**Review by mariajgrimaldi — APPROVED** [review]
Since this has backward compatibility if a new incompatible variable is configured then the sentry configuration will default to the old configuration (without extra option compatibilities). So this is technically safe to merge!

**Pull request merged**